tools/clang/test/HLSLFileCheck/hlsl/intrinsics/cast/asdoublefold.hlsl

```diff
@@ -0,0 +1,14 @@
+// RUN: %dxc -E main -T ps_6_0 %s | FileCheck %s
+
+// Verify that constants provided to asdouble can be folded
+// CHECK-NOT: makeDouble
+// CHECK: splitDouble.f64(i32 102, double -1.000000e+00
+
+
+uint4 main() : SV_Target
+{
+  double d = asdouble(0, 0xBFF00000);
+  uint4 ret;
+  asuint(d, ret.xy, ret.zw);
```

Comment:
Strange that output dim for low/high here is vec2 when input is scalar?  This produces splat of low into `ret.xy` and splat of high into `ret.zw`.  Is that what the code intended?